pandas/tests/frame/test_analytics.py

```diff
@@ -1842,6 +1842,16 @@ def test_numpy_round(self):
         with pytest.raises(ValueError, match=msg):
             np.round(df, decimals=0, out=df)
 
+    @pytest.mark.skipif(
+        PY2 and is_platform_windows(), reason="numpy/numpy#7882")
+    def test_numpy_round_nan(self):
+        # See gh-14197
+        df = Series([1.53, np.nan, 0.06]).to_frame()
+        with np.errstate(invalid='raise'):
```

Comment:
can u add assert_produces_warning() to these tests asserting that no warning happens

Reply:
i don't think there is any value add. the RuntimeWarning issued by numpy seems to be only issued once and therefore is highly unlikely to be detected by a test unless run in isolation. hence the `with np.errstate(invalid='raise'):` context mangaer is used to convert the `RuntimeWarning: invalid value encountered in rint` to `FloatingPointError: invalid value encountered in rint`. Also `assert_produces_warning` is not thread-safe.